### DISCUSSION TITLE
Should exist behaving oddly?

### DISCUSSION THREAD

**Discussion by pcgeek86** [pester/Pester#2244] [Q&A] [answered]
I have the following Pester test, which is failing. Anyone know how to rewrite this test so that it succeeds? How do I properly use the `Should Exist` command?

```
Describe DNS {
  it 'Google DNS should be accessible' {
    (Resolve-DnsName -Server 8.8.8.8 -Name trevorsullivan.net) | Should Exist
  }
}
```

### Actual Result

```
 [-] Google DNS should be accessible 113ms
   Expected: {Microsoft.DnsClient.Commands.DnsRecord_A} to exist
   4:     (Resolve-DnsName -Server 8.8.8.8 -Name trevorsullivan.net) | Should Exist
   at <ScriptBlock>, <No file>: line 4
```

### Expected Result

Test passes.

**Comment by fflaten** [comment, marked as answer]
`Should -Exists` is for testing paths, ex. a filesystem path. See https://pester.dev/docs/assertions/#exist

In your example, you'd probably want to use `Should -Not -BeNullOrEmpty`, or save the response in a variable and test one or more properties (count, IP, record types etc.) using `Should -Be`, `Should -HaveCount` etc.

**Comment by pcgeek86** [comment]
Thanks, I also tried `Should Not Throw` and that seemed to work okay.

**Reply by fflaten** [reply]
Also an option, but remember:
- `Resolve-DnsName` doesn't throw a terminating error (required for `-Throw` to work) by default. Use `-ErrorAction Stop`.
- `Should -Throw` expects a scriptblock unlike the other assertions

Ex: `{ Resolve-DnsName -Server 8.8.8.8 -Name trevorsullivan.net -ErrorAction Stop } | Should -Not -Throw`.
